source/common/network/utility.cc

```diff
@@ -368,13 +401,23 @@ const std::string& Utility::getIpv6CidrCatchAllAddress() {
 
 Address::InstanceConstSharedPtr Utility::getAddressWithPort(const Address::Instance& address,
                                                             uint32_t port) {
+  Envoy::StatusOr<Address::InstanceConstSharedPtr> addr;
   switch (address.ip()->version()) {
   case Address::IpVersion::v4:
-    return std::make_shared<Address::Ipv4Instance>(address.ip()->addressAsString(), port);
+    addr = Address::InstanceFactory::createInstancePtr<Address::Ipv4Instance>(
+        address.ip()->addressAsString(), port);
+    break;
   case Address::IpVersion::v6:
-    return std::make_shared<Address::Ipv6Instance>(address.ip()->addressAsString(), port);
+    addr = Address::InstanceFactory::createInstancePtr<Address::Ipv6Instance>(
+        address.ip()->addressAsString(), port);
+    break;
+  default:
+    NOT_REACHED_GCOVR_EXCL_LINE;
   }
-  NOT_REACHED_GCOVR_EXCL_LINE;
+  if (!addr.ok()) {
+    return nullptr;
```

Comment:
previously an error here would have resulted in a Throw, right? Should we return the StatusOr here so that callers can find the error message where they previously had caught an error and logged it in context?

In general you should make sure your changes don't drop information that may be needed when debugging a crash with logs.